tests/test_general.py

```diff
@@ -68,6 +69,68 @@ def test_tilt_desired(self):
         ].empty
 
 
+class TestRTLHeight:
+    # The return to home height changes with the distance from home
+    # mode was triggered
+    # check the height above ground while the drone returns to home. compare it with 
+    # the allowed maximum or minimum heights, until the drone has reached home and motors have been turned off
+
+   def setup_class(self):
+        # get the required data
+        filepath = getfilepath()
+        topics = [
+            "vehicle_local_position",
+            "vehicle_status",
+        ]
+        self.ulog = pyulog.ULog(filepath, topics)
+        self.df = ulogconv.merge(ulogconv.createPandaDict(self.ulog))
+
+
+   def test_rtl(self):
+       # TODO adjust this test to also work with cone (only for yuneec)
+        NAVIGATION_STATE_AUTO_RTL = 5           # see https://github.com/PX4/Firmware/blob/master/msg/vehicle_status.msg
+        max_height_within_RTL_MIN_DIST = 10     # within this radius, the drone should not go higher than this value after RTL has been triggered
+        thresh = 1                              # Threshold for position inaccuracies
+
+        # drone parameters: below rtl_min_dist, the drone follows different rules than outside of it.
+        rtl_min_dist = (
+            loginfo.get_param(self.ulog, "RTL_MIN_DIST", 0)
+        )
+        rtl_return_alt = (
+            loginfo.get_param(self.ulog, "RTL_RETURN_ALT", 0)
+        )
+
+        posanl.add_horizontal_distance(self.df)
+
+        # Check vehicle status for RTL trigger
+        # store all distance and height values that are reached ONLY DURING RTL in a new array
+        distance_after_RTL = []
+        height_after_RTL = []
+        for i in range(self.df.T_vehicle_status_0__F_nav_state.count() - 1):
+            # at least two consecutive T_vehicle_status_0__F_nav_state values have to be equal to NAVIGATION_STATE_AUTO_RTL in order to confirm that RTL has been triggered
+            # TODO if RTL was triggered more than once during a flight, look for the maximum height for every time RTL was triggered
```

Comment:
In order to implement this logic where two consecutive state values have to be present and each RTL trigger is handled independent, I would use pandas groupby:

1. create a new column that groups `T_vehicle_status_0__F_nav_state` into groups along the timeline, where a group contains the same consecutive value. for instance, the array [1,1,3,4,1,3,3] would have the following groups: [1,1], [3], [4],[1],[3,3]
```bash
self.df['T_vehicle_status_0__F_nav_state_group2'] = (df.T_vehicle_status_0__F_nav_state != self.df.T_vehicle_status_0__F_nav_state .shift()).cumsum()
```
2. create groups based on that new field
```bash
group = self.df.groupby(['T_vehicle_status_0__F_nav_state_group2'])
```

3. loop through all groups and check count > 1 as test criteria:
```bash
for g, d in group:
     if d.T_vehicle_status_0__F_nav_state.count() > 1:
         `TEST`
```